src/main/java/billgates/view/gui/AddEntryDialog.java

```diff
@@ -1,112 +1,108 @@
-            this.to = toField.getText();
-            this.location = locationField.getText();
+            LocalDateTime datetime = this.dateTimePicker.getDateTimePermissive();
+            InsertEntryRequestModel model = new InsertEntryRequestModel(
+                    ZonedDateTime.of(datetime, ZoneId.systemDefault()),
+                    Double.parseDouble(this.valueField.getText()),
+                    Objects.requireNonNull(this.currencyComboBox.getSelectedItem()).toString(),
+                    this.descriptionField.getText(),
+                    this.fromField.getText(),
+                    this.toField.getText(),
+                    this.locationField.getText());
+            if (this.mainFrame.isSplitterBill()) {
+                model.setPayee(this.payeeField.getText());
+                model.setPaidBack(this.paidBackCheckBox.isSelected());
+            }
+            return model;
+        } else {
+            return null;
         }
     }
 
-    public ZonedDateTime getDate() {
-        return date;
-    }
-
-    public double getValue() {
-        return value;
-    }
-
-    public String getCurrency() {
-        return currency;
-    }
-
-    public String getDescription() {
-        return description;
-    }
-
-    public String getFrom() {
-        return from;
-    }
-
-    public String getTo() {
-        return to;
-    }
-
-    public String getLocationText() {
-        return location;
-    }
-
-    public int getResult() {
-        return result;
+    private void initDialog() {
+        this.add(new ActionLabel("Date and Time:"));
+        // Create a datetime picker.
+        DatePickerSettings datePickerSettings = new DatePickerSettings();
+        TimePickerSettings timePickerSettings = new TimePickerSettings();
+        timePickerSettings.setFormatForMenuTimes(BillGatesUtilities.TIME_PATTERN);
+        timePickerSettings.setFormatForDisplayTime(BillGatesUtilities.TIME_PATTERN);
+        this.dateTimePicker = new DateTimePicker(datePickerSettings, timePickerSettings);
+        this.dateTimePicker.setAlignmentX(LEFT_ALIGNMENT);
+        this.dateTimePicker.setDateTimeStrict(LocalDateTime.now());
+        this.add(this.dateTimePicker);
+
+        this.add(new ActionLabel("Value:"));
+        this.valueField = new ActionTextField(20);
+        this.valueField.setDocument(new RegexDocument("\\d*\\.?\\d{0,2}"));
+        this.valueField.setText(NumberFormat.getInstance().format(0D));
+        this.add(this.valueField);
+
+        this.add(new ActionLabel("Currency:"));
+        this.currencyComboBox = new ActionComboBox<>(BillGatesUtilities.CURRENCY_CODES);
+        this.currencyComboBox.setAlignmentX(LEFT_ALIGNMENT);
+        this.currencyComboBox.setFont(ActionTextField.DEFAULT_FONT);
+        this.add(this.currencyComboBox);
+
+        this.add(new ActionLabel("Description:"));
+        this.descriptionField = new ActionTextField(20);
+        this.add(this.descriptionField);
```

Comment:
Do we not want to `this.___.setAlignmentX(LEFT_ALIGNMENT)`on everything else?

![image](https://user-images.githubusercontent.com/8016180/205577663-63e6033b-9032-4a75-8843-b887aa389021.png)


The right hand side is the example where I added `LEFT_ALIGNMENT` to everything, looks a little bit cleaner IMO?